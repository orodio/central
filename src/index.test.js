import createStore from './index'
import { Map } from 'immutable'

const snap = (msg, value) =>
  test(msg, () => expect(value).toMatchSnapshot())

describe('createStore', () => {
  const {
    handle,
    $,
    intent,
    getState,
    dispatch,
    subscribe,
    connect,
  } = createStore(new Map({
    count: 5,
  }))

  handle('inc_by', (state, delta = 1) => state.updateIn(['count'], count => count + delta))
  intent('inc', () => dispatch('inc_by'))
  intent('inc_10', () => dispatch('inc_by', 10))

  const inc = $('inc')
  const inc10 = $('inc_10')

  snap("initial state", getState(['count']))
  dispatch("inc_by")
  snap("after raw dispatch of inc_by", getState(['count']))
  inc()
  snap("after intent thunk inc", getState(['count']))
  inc10()
  snap('after intent thunk inc10', getState(['count']))

  const Counter = ({ count }) =>
    <div id="Counter">
      { count }
      <button id="inc" onClick={() => inc10()}>+</button>
    </div>

  const ConnectedCounter = connect((state) => ({
    count: state(['count'], 0)
  }))(Counter)

  const Comp = mount(<ConnectedCounter/>)
  snap('initial component', Comp)

  Comp.find('#inc').simulate('click')
  snap('after click of inc button', Comp)
})
