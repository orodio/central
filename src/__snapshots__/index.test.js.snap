// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createStore after click of inc button 1`] = `
<Component>
  <Connected>
    <Counter
      count={27}
    >
      <div
        id="Counter"
      >
        27
        <button
          id="inc"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </Counter>
  </Connected>
</Component>
`;

exports[`createStore after intent thunk inc 1`] = `7`;

exports[`createStore after intent thunk inc10 1`] = `17`;

exports[`createStore after raw dispatch of inc_by 1`] = `6`;

exports[`createStore initial component 1`] = `
<Component>
  <Connected>
    <Counter
      count={27}
    >
      <div
        id="Counter"
      >
        27
        <button
          id="inc"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </Counter>
  </Connected>
</Component>
`;

exports[`createStore initial state 1`] = `5`;
